File: pkg/codegen/testing/utils/testdata/output-funcs-aws-pp/nodejs/output-funcs-aws.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const aws_vpc = new aws.ec2.Vpc("aws_vpc", {
    cidrBlock: "10.0.0.0/16",
    instanceTenancy: "default",
});
const privateS3VpcEndpoint = new aws.ec2.VpcEndpoint("privateS3VpcEndpoint", {
    vpcId: aws_vpc.id,
    serviceName: "com.amazonaws.us-west-2.s3",
});
const privateS3PrefixList = aws.ec2.getPrefixListOutput({
    prefixListId: privateS3VpcEndpoint.prefixListId,
});
const bar = new aws.ec2.NetworkAcl("bar", {vpcId: aws_vpc.id});
const privateS3NetworkAclRule = new aws.ec2.NetworkAclRule("privateS3NetworkAclRule", {
    networkAclId: bar.id,
    ruleNumber: 200,
    egress: false,
    protocol: "tcp",
    ruleAction: "allow",
    cidrBlock: privateS3PrefixList.cidrBlocks[0],
    fromPort: 443,
    toPort: 443,
});
// A contrived example to test that helper nested records ( `filters`
// below) generate correctly when using output-versioned function
// invoke forms.
const amis = aws.ec2.getAmiIdsOutput({
    owners: [bar.id],
    filters: [{
        name: bar.id,
        values: ["pulumi*"],
    }],
});

File: pkg/codegen/testing/utils/testdata/output-funcs-aws-pp/python/output-funcs-aws.py
import pulumi
import pulumi_aws as aws

aws_vpc = aws.ec2.Vpc("aws_vpc",
    cidr_block="10.0.0.0/16",
    instance_tenancy="default")
private_s3_vpc_endpoint = aws.ec2.VpcEndpoint("privateS3VpcEndpoint",
    vpc_id=aws_vpc.id,
    service_name="com.amazonaws.us-west-2.s3")
private_s3_prefix_list = aws.ec2.get_prefix_list_output(prefix_list_id=private_s3_vpc_endpoint.prefix_list_id)
bar = aws.ec2.NetworkAcl("bar", vpc_id=aws_vpc.id)
private_s3_network_acl_rule = aws.ec2.NetworkAclRule("privateS3NetworkAclRule",
    network_acl_id=bar.id,
    rule_number=200,
    egress=False,
    protocol="tcp",
    rule_action="allow",
    cidr_block=private_s3_prefix_list.cidr_blocks[0],
    from_port=443,
    to_port=443)
# A contrived example to test that helper nested records ( `filters`
# below) generate correctly when using output-versioned function
# invoke forms.
amis = aws.ec2.get_ami_ids_output(owners=[bar.id],
    filters=[{
        "name": bar.id,
        "values": ["pulumi*"],
    }])
